lm_eval/base.py

```diff
@@ -342,32 +339,62 @@ def _collate(x):
 
         re_ord = utils.Reorderer(requests, _collate)
 
-        for context, until in tqdm(re_ord.get_reordered()):
+        for request in tqdm(re_ord.get_reordered()):
+            if len(request) == 2:
+                # Unpack greedy sample request
+                context, until, = request
+                k, temperature = 1, 0.
+                greedy = True
+                _model_generate_kwargs = {}
+            elif len(request) == 4:
+                # Unpack temperature sample request
+                context, until, k, temperature = request
+                for key in ["k", "temperature"]:
+                    assert key in inspect.getfullargspec(self._model_generate).args, \
+                        f"Model generation parameter '{key}' not accepted as an argument for _model_generate"
+                greedy = False
+                _model_generate_kwargs = {"k": k, "temperature": temperature}
+            elif len(request) == 5:
+                context, until, k, temperature, k_batch = request
```

Comment:
Perhaps name `k` and `k_batch` to line up with what is specified in the config? E.g. right now the config has `majority_voting` (or rename to `majority_voting_k` or `majority_voting_num_votes`). 

This could help with keeping track of (a) which hyperparameters are supported, and (b) which part of the code corresponds to what is specified in the config

Reply:
I think that this is a very good suggestion. Improving consistency of internal names for things is a huge add for hackability.

Reply:
Agreed- after thinking about it some more, the `k` here isn't specific to majority voting though, so perhaps just `num_samples`